src/lib/zf/log.c

```diff
@@ -135,14 +136,44 @@ void zf_log(struct zf_stack* st, const char* fmt, ...)
   va_end(va);
 }
 
+static int zf_log_set_file_name(const char *file)
+{
+  if( strnlen(file, ZF_LOG_FILE_NAME_SIZE) > ZF_LOG_FILE_NAME_SIZE-1 ) {
+    zf_log_stack_err(NO_STACK, "%s: File name is longer than %i characters: %s\n",
+                     __func__, ZF_LOG_FILE_NAME_SIZE-1, strerror(-EINVAL));
+    return -EINVAL;
+  }
+
+  strncpy(zf_log_file_name, file, sizeof(zf_log_file_name));
+  zf_log_file_name[ZF_LOG_FILE_NAME_SIZE-1] = '\0';
+
+  return 0;
+}
+
 int zf_log_redirect(const char* file)
 {
-  FILE* f = fopen(file, "a");
-  if( ! f )
+  char* previous_file_name = strdup(zf_log_file_name);
+  if( ! previous_file_name )
     return -errno;
 
+  int rc = zf_log_set_file_name(file);
+  if( rc < 0 ) {
+    free(previous_file_name);
+    return rc;
+  }
+
+  FILE* f = fopen(zf_log_file_name, "a");
+  if( ! f ) {
+    strncpy(zf_log_file_name, previous_file_name, sizeof(zf_log_file_name));
```

Comment:
```suggestion
  if( !f ) {
    zf_log_set_file_name(previous_file_name)
```

Prefer function reuse despite redundant checks